sdcm/utils/adaptive_timeouts/load_info_store.py

```diff
@@ -251,6 +253,54 @@ def get(self, operation: str | None, timeout_occurred: bool | None = None):
         return [hit["_source"] for hit in res["hits"]["hits"]]
 
 
+class ArgusAdaptiveTimeoutStore(AdaptiveTimeoutStore):
+    """
+    Report adaptive timeout results to Argus.
+    """
+
+    def __init__(self):
+        self.test_config = TestConfig()
+
+    def send_adaptive_timeout_results_to_argus(self, results: dict):
+        argus_client = self.test_config.argus_client()
+        if not argus_client:
+            LOGGER.warning("Will not submit to argus - no client initialized")
+            return
+
+        class AdaptiveTimeoutResultsTable(GenericResultTable):
```

Comment:
Could we make this a normal class instead so we could reuse it? I dont think we should be using inner classes.

Reply:
we are not gonna reuse it, and if we do, we can move it.
we are doing this in multiple places, since the dynamic nature those tables

Reply:
We want dynamic tables but we are using static classes to implement that, is there a reason why we implement it like this in the first place?

This can be addressed in the followup PR, but but I would like to understand the design here

Reply:
it's following the same pattern of usage of `GenericResultTable`, those tables are not share across multiple places, hence it's doesn't matter if they are defined on their own, or inline.

if for whatever reason it would be needed on it's own, it can be moved.

Reply:
`GenericResultTable.as_dict()` is only caring about values in `self`, so we can easily make this class dynamic (without changing anything else) by using init function and not having it done through Meta

Reply:
I tested it locally and it cannot be done easily so closing this comments. There is actual logic tied to the Meta class, which in opinion is wrong and prevents better usage patterns.